rkt/image/fetcher.go

```diff
@@ -79,12 +112,17 @@ func (f *Fetcher) getAsc(ascPath string) *asc {
 // fetchImageDeps will recursively fetch all the image dependencies
 func (f *Fetcher) fetchImageDeps(hash string) error {
 	imgsl := list.New()
-	seen := map[string]struct{}{}
+	seen := map[dist.Distribution]struct{}{}
```

Comment:
this seems very dangerous to me as it is very easy to miss the equality invariant of the underlying dynamic type. I'd rather make this `map[string]dist.Distrbution`, where the key is the string representation of the cimd URL.